sickbeard/subtitles.py

```diff
@@ -565,68 +798,93 @@ def dhm(td):
+                           show_indexerid):
+    """Executes the subtitles extra-scripts for the given video path
+
+    :param video_path: the video path
+    :type video_path: str
+    :param subtitle_path: the downloaded subtitle path
+    :type subtitle_path: str
+    :param subtitle_language: the subtitle language
+    :type subtitle_language: babelfish.Language
+    :param show_name: the show name
+    :type show_name: str
+    :param season: the episode season number
+    :type season: int
+    :param episode: the episode number
+    :type episode: int
+    :param episode_name: the episode name
+    :type episode_name: str
+    :param show_indexerid: the show indexer id
+    :type show_indexerid: int
     """
+    run_subs_scripts(sickbeard.SUBTITLES_EXTRA_SCRIPTS,
+                     [video_path, subtitle_path, subtitle_language.opensubtitles, show_name, str(season), str(episode),
+                      episode_name, str(show_indexerid)])
 
-    subtitle_scripts = sickbeard.SUBTITLES_PRE_SCRIPTS if is_pre else sickbeard.SUBTITLES_EXTRA_SCRIPTS
-    for script_name in subtitle_scripts:
+
+def run_subs_scripts(scripts, script_args):
+    """Execute subtitle scripts
+
+    :param scripts: the script commands to be executed
+    :type scripts: list of str
+    :param script_args: the arguments to be passed to the script
+    :type script_args: list of str
+    """
+    for script_name in scripts:
         script_cmd = [piece for piece in re.split("( |\\\".*?\\\"|'.*?')", script_name) if piece.strip()]
 
-        if is_pre:
-            logger.log(u'Running subtitle pre-script: {}'.format(script_name))
-            inner_cmd = script_cmd + [video_filename]
-        else:
-            logger.log(u'Running subtitle extra-script: {}'.format(script_name))
-            inner_cmd = script_cmd + [video_filename, subtitle_path, subtitle_language.opensubtitles,
-                                      subtitles_info['show_name'], str(subtitles_info['season']),
-                                      str(subtitles_info['episode']), subtitles_info['name'],
-                                      str(subtitles_info['show_indexerid'])]
+        logger.log(u'Running subtitle {0}-script: {1}'.format('extra' if len(script_args) > 1 else 'pre', script_name))
+        inner_cmd = script_cmd + script_args
 
         # use subprocess to run the command and capture output
-        logger.log(u'Executing command: {}'.format(inner_cmd))
+        logger.log(u'Executing command: {0}'.format(inner_cmd))
         try:
             process = subprocess.Popen(inner_cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
                                        stderr=subprocess.STDOUT, cwd=sickbeard.PROG_DIR)
             out, _ = process.communicate()  # @UnusedVariable
-            logger.log(u'Script result: {}'.format(out), logger.DEBUG)
+            logger.log(u'Script result: {0}'.format(out), logger.DEBUG)
```

Comment:
@ratoaq2 this fixes the AA:

`logger.log(u'Script result: {0}'.format(out.strip()), logger.DEBUG)`